dowhy/causal_refuters/bootstrap_refuter.py

```diff
@@ -47,6 +89,29 @@ def refute_estimate(self, *args, **kwargs):
                                     n_samples=self._sample_size,
                                     random_state=self._random_state )
 
+            if self._chosen_variables is not None:
+                for variable in self._chosen_variables:
+                    
+                    if ('float' or 'int') in new_data[variable].dtype.name: 
+                        new_data[variable] += np.random.randn(self._sample_size) * self._noise
+                    
+                    elif 'bool' in new_data[variable]:
+                        change_mask = self.get_mask_variables()
+                        # Set these values to zero
+                        new_data[variable] *= change_mask
+                        # Sample a binomial distribution and set the value for those datapoints we set to zero in the previous step
+                        new_data[variable] += (1 - change_mask) * np.random.binomial(BootstrapRefuter.DEFAULT_NUMBER_OF_TRIALS,

```

Comment:
This is not needed for the bool type. How about simply using a "probability of change" as a parameter, and then changing the flipping the value of those rows whose random number falls below that parameter?